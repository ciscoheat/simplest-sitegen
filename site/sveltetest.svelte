<script>
	export const title = "Svelte test";
	let m = { x: 0, y: 0 };

	function handleMousemove(event) {
		m.x = event.clientX;
		m.y = event.clientY;
	}
</script>

<h2>{title}</h2>

<div class="box" on:mousemove={handleMousemove}>
	<div class="text">The mouse is at {m.x} x {m.y}</div>
</div>

<style>
	.box { 
		width: 100%; height: 400px; 
		background-color: aquamarine;
	}

	.box .text {
		font-style: italic;
	}
</style>