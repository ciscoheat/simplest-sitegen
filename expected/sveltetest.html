<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Svelte test</title>
  <link rel="stylesheet" href="/CSS/chota.min.css?glxzhr">
  <link rel="stylesheet" href="/CSS/fulloo.css?1dpkm1x">
  
</head>
<body>
  <nav class="nav hide-pr" id="tabs">
    <div class="nav-center">
      <a href="/">Home</a>
      <a href="/Introduction/">Introduction</a>
      <a href="/FAQ/">FAQ</a>
      <a href="/Documents/">Documents</a>
      <a href="/Examples/">Examples</a>
      <a href="/Downloads/">Downloads</a>
      <a href="/Events/">Events</a>
      <a href="/oowiki.php">Wiki</a>
    </div>
  </nav>

  <main class="container">
    
        <style>.box.svelte-tcxy2u.svelte-tcxy2u{width:100%;height:400px;background-color:aquamarine}.box.svelte-tcxy2u .text.svelte-tcxy2u{font-style:italic}</style>
        <div id="simplest-svelte-component"><h2>Svelte test</h2>

<div class="box svelte-tcxy2u"><div class="text svelte-tcxy2u">The mouse is at 0 x 0</div>
</div></div>
        <script type="module">import{SvelteComponent as t,append_hydration as e,attr as s,children as c,claim_element as o,claim_space as l,claim_text as n,detach as r,element as i,init as a,insert_hydration as h,listen as u,noop as m,safe_not_equal as x,set_data as v,space as y,text as d}from"/js/svelte.js?y5l7e1";function f(t){let a,x,f,p,E,g,I,j,D,T,V,$=t[1].x+"",b=t[1].y+"";return{c(){a=i("h2"),x=d(t[0]),f=y(),p=i("div"),E=i("div"),g=d("The mouse is at "),I=d($),j=d(" x "),D=d(b),this.h()},l(e){a=o(e,"H2",{});var s=c(a);x=n(s,t[0]),s.forEach(r),f=l(e),p=o(e,"DIV",{class:!0});var i=c(p);E=o(i,"DIV",{class:!0});var h=c(E);g=n(h,"The mouse is at "),I=n(h,$),j=n(h," x "),D=n(h,b),h.forEach(r),i.forEach(r),this.h()},h(){s(E,"class","text svelte-tcxy2u"),s(p,"class","box svelte-tcxy2u")},m(s,c){h(s,a,c),e(a,x),h(s,f,c),h(s,p,c),e(p,E),e(E,g),e(E,I),e(E,j),e(E,D),T||(V=u(p,"mousemove",t[2]),T=!0)},p(t,[e]){2&e&&$!==($=t[1].x+"")&&v(I,$),2&e&&b!==(b=t[1].y+"")&&v(D,b)},i:m,o:m,d(t){t&&r(a),t&&r(f),t&&r(p),T=!1,V()}}}function p(t,e,s){let c={x:0,y:0};return["Svelte test",c,function(t){s(1,c.x=t.clientX,c),s(1,c.y=t.clientY,c)}]}new class Component extends t{constructor(t){super(),a(this,t,p,f,x,{title:0})}get title(){return this.$$.ctx[0]}}({target:document.getElementById("simplest-svelte-component"),hydrate:!0});</script>
      
  </main>
  
  <script src="/js/highlight.min.js"></script>
  <script>
    // Navigation highlight
    const current = document.location.pathname
    document.querySelectorAll('#tabs a').forEach(a => {
      const tab = new URL(a.href).pathname

      if(tab == current) {
        a.classList.add('active')
      }
    })
  </script>    
</body>
</html>
